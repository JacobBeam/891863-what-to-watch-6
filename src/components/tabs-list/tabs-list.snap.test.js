import React from 'react';
import {render} from '@testing-library/react';
import {createMemoryHistory} from 'history';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {Router} from 'react-router-dom';
import TabsList from './tabs-list';

const api = jest.fn();
const middlleware = thunk.withExtraArgument(api);
const mockStore = configureStore([middlleware]);

it(`Should TabsList render correctly`, () => {
  const history = createMemoryHistory();
  const store = mockStore({
    FILM_DATA: {films: [],
      selectedMovie: {
        "starring": [],
      },
    },
  });

  const {container} = render(
      <redux.Provider store={store}>
        <Router history={history}>
          <TabsList/>
        </Router>
      </redux.Provider>
  );
  expect(container).toMatchSnapshot();
});
