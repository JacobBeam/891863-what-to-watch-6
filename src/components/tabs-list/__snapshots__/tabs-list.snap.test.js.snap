// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should TabsList render correctly 1`] = `
<div>
  <div
    class="movie-card__desc"
  >
    <nav
      class="movie-nav movie-card__nav"
    >
      <ul
        class="movie-nav__list"
      >
        <li
          class="movie-nav__item movie-nav__item--active"
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Overview
          </a>
        </li>
        <li
          class="movie-nav__item "
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Details
          </a>
        </li>
        <li
          class="movie-nav__item "
        >
          <a
            class="movie-nav__link"
            href="#"
          >
            Reviews
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="movie-rating"
    >
      <div
        class="movie-rating__score"
      />
      <p
        class="movie-rating__meta"
      >
        <span
          class="movie-rating__level"
        />
        <span
          class="movie-rating__count"
        >
           ratings
        </span>
      </p>
    </div>
    <div
      class="movie-card__text"
    >
      <p />
      <p
        class="movie-card__director"
      >
        <strong>
          Director: 
        </strong>
      </p>
      <p
        class="movie-card__starring"
      >
        <strong />
      </p>
    </div>
  </div>
</div>
`;
